File: crates/auto-lang/test/a2c/006_struct/struct.expected.c
#include <stdio.h>

struct Point {
    int x;
    int y;
};

int main(void) {
    struct Point p = {.x = 1, .y = 2};
    p.x = 3;
    printf("%s %d %s %d\n", "P: ", p.x, ", ", p.y);
    return 0;
}

File: crates/auto-lang/test/a2ts/02_types/001_struct/struct.expected.ts
export class Point {
    x: number;
    y: number;

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }
}

export class Circle {
    radius: number;
    border: number;
    center: Point;

    constructor(radius: number, border: number, center: Point) {
        this.radius = radius;
        this.border = border;
        this.center = center;
    }
}

function main(): void {
    let p = new Point(1, 2);
    p.x = 3;
    console.log(`P: ${p.x}, ${p.y}`);
    

    const circle = new Circle(5, 1, new Point(50, 50));
    console.log(`C: ${circle.center.x}, ${circle.center.y}, ${circle.radius}`);
}

main();
